File: app/src/main/java/com/shen/xi/android/tut/DisclaimerFragment.java
package com.shen.xi.android.tut;


import android.os.Bundle;
import android.preference.PreferenceManager;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentTransaction;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.google.inject.Inject;
import com.squareup.otto.Bus;

/**
 * A fragment with a Google +1 button.
 * Use the {@link DisclaimerFragment#newInstance} factory method to
 * create an instance of this fragment.
 */
public class DisclaimerFragment extends Fragment {

    @Inject
    private Bus mBus;

    public DisclaimerFragment() {
        TuTModule.getInjector().injectMembers(this);
    }

    /**
     * Use this factory method to create a new instance of
     * this fragment using the provided parameters.
     *
     * @return A new instance of fragment DisclaimerFragment.
     */
    public static DisclaimerFragment newInstance() {
        return new DisclaimerFragment();
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_disclaimer, container, false);

        view.findViewById(android.R.id.button1).setOnClickListener(
                new View.OnClickListener() {
                    @Override
                    public void onClick(View view) {
                        PreferenceManager.getDefaultSharedPreferences(getActivity())
                                .edit()
                                .putBoolean(MainActivity.PREF_DISCLAIMER_AGREE, true)
                                .commit();

                        getFragmentManager().beginTransaction()
                                .setTransition(FragmentTransaction.TRANSIT_FRAGMENT_CLOSE)
                                .remove(DisclaimerFragment.this)
                                .commit();
                    }
                }
        );

        return view;
    }

}

File: app/src/main/scala/com/shen/xi/android/tut/DisclaimerFragment.scala
package com.shen.xi.android.tut


import android.os.Bundle
import android.preference.PreferenceManager
import android.support.v4.app.{Fragment, FragmentTransaction}
import android.view.{LayoutInflater, View, ViewGroup}
import com.google.inject.Inject
import com.squareup.otto.Bus


object DisclaimerFragment {
  def newInstance() = new DisclaimerFragment()
}

/**
 * A fragment with a Google +1 button.
 * Use the 'DisclaimerFragment#newInstance' factory method to
 * create an instance of this fragment.
 */
class DisclaimerFragment extends Fragment {

  @Inject
  private var mBus: Bus = null

  TuTModule.getInjector.injectMembers(this)

  /**
   * Use this factory method to create a new instance of
   * this fragment using the provided parameters.
   *
   * @return A new instance of fragment DisclaimerFragment.
   */
  override def onCreateView(inflater: LayoutInflater, container: ViewGroup, savedInstanceState: Bundle) = {
    // Inflate the layout for this fragment
    val view = inflater.inflate(R.layout.fragment_disclaimer, container, false)

    view.findViewById(android.R.id.button1).setOnClickListener(
      new View.OnClickListener() {
        override def onClick(view: View) = {
          PreferenceManager.getDefaultSharedPreferences(getActivity)
            .edit()
            .putBoolean(MainActivity.PREF_DISCLAIMER_AGREE, true)
            .commit()

          getFragmentManager
            .beginTransaction()
            .setTransition(FragmentTransaction.TRANSIT_FRAGMENT_CLOSE)
            .remove(DisclaimerFragment.this)
            .commit()
        }
      }
    )

    view
  }

}
